src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        result += separator;
+        continue;
+      }
+
+      if (i !== this.length - 1) {
+        result += this[i].toString() + separator;
```

Comment:
```suggestion
        result += this[i] + separator;
```
If you add element to a string, it becomes string